src/client/components/stockDataRegion/index.js

```diff
@@ -3,7 +3,8 @@ import { connect } from 'react-redux';
 import StockDataRegion from './StockDataRegion';
 
 const mapStateToProps = state => ({
-  hasData: state.selectedStock !== ''
+  showNoDataMessage: !state.search.lastSearch && !state.selectedStock,
```

Comment:
Isn't !state.selectedStock sufficient? Is selectedStock ever going to be empty when lastSearch isn't?
Also lastSearch has been removed in my other branch assuming when you reviewed that PR you were okay with it.

Reply:
yes, !state.fetching will be a viable replacement, and my PR is in.
good luck